application/models/message.go

```diff
@@ -192,3 +192,24 @@ func (m *Message) FindByUUID(id string) error {
 
 	return nil
 }
+
+// FindByUserAndUUID loads from DB the Message record identified by the given UUID, if the given user is allowed.
+func (m *Message) FindByUserAndUUID(user User, id string) error {
+	if err := m.FindByUUID(id); err != nil {
```

Comment:
I suppose`FindByUUID` shouldn't be exported anymore, so we don't accidentally use it instead of `FindByUserAndUUID`.